zooniverse/front-end-monorepo
# lib-user: Fix BarChart console error, conditionally define gradient per data

## Package
- lib-user

## Describe your changes
- if data is an empty array, the `Math.max` calculation yielded `-Infinity` and caused a console error when Grommet's DataChart attempted to create the chart color gradient
- these changes define a default `gradient` of `'brand'` for empty data, then if data defines the `gradient` based on the data max value

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - see error:
    - go to your user stats page or a group stats page and note console error `Error: <stop> attribute offset: Expected number or percentage, "NaN".`
  - see fix:
    - from app-root, run `yarn dev` for staging (or `yarn build`, then `yarn start` for production), go to your staging/production user stats page or a group stats page and note no related console error
- Which storybook stories should be reviewed? n/a
- Are there plans for follow up PR’s to further fix this bug or develop this feature? no

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## Bug Fix
